app/lib/actions/action_with_sub_plans.rb

```diff
@@ -0,0 +1,156 @@
+module Actions
+
+  class Actions::ActionWithSubPlans < Actions::EntryAction
+
+    middleware.use Actions::Middleware::KeepCurrentUser
+    middleware.use Actions::Middleware::SetCurrentTaskId
+
+    SubPlanFinished = Algebrick.type do
+      fields! :execution_plan_id => String,
+              :success           => type { variants TrueClass, FalseClass }
+    end
+
+    def plan(*args)
+      raise NotImplementedError
+    end
+
+    def run(event = nil)
+      case(event)
+      when nil
+        if output[:total_count]
+          resume
+        else
+          sub_plans = create_sub_plans
+          sub_plans = Array[sub_plans] unless sub_plans.is_a? Array
```

Comment:
You can't do this, as the trigger result is an algerbick object, implementing `to_a` method